mattermost-plugin/webapp/src/index.tsx

```diff
@@ -80,21 +80,33 @@ export default class Plugin {
     // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
     public async initialize(registry: PluginRegistry, store: Store<GlobalState, Action<Record<string, unknown>>>) {
         this.registry = registry
-        const goToFocalboardWorkspace = () => {
-            const currentChannel = store.getState().entities.channels.currentChannelId
-            window.open(`${window.location.origin}/boards/workspace/${currentChannel}`)
-        }
-        this.channelHeaderButtonId = registry.registerChannelHeaderButtonAction(<FocalboardIcon/>, goToFocalboardWorkspace, '', 'Focalboard Workspace')
-
-        this.registry.registerCustomRoute('go-to-current-workspace', () => {
-            const history = useHistory()
-            const currentChannel = store.getState().entities.channels.currentChannelId
-            history.push(`/boards/workspace/${currentChannel}`)
-            return <></>
-        })
 
         if (this.registry.registerProduct) {
+            const goToFocalboardWorkspace = () => {
+                const currentChannel = store.getState().entities.channels.currentChannelId
+                window.open(`${window.location.origin}/boards/workspace/${currentChannel}`)
+            }
+            this.channelHeaderButtonId = registry.registerChannelHeaderButtonAction(<FocalboardIcon/>, goToFocalboardWorkspace, '', 'Focalboard Workspace')
+
+            this.registry.registerCustomRoute('go-to-current-workspace', () => {
+                const history = useHistory()
+                useEffect(() => {
+                    const currentChannel = store.getState().entities.channels.currentChannelId
+                    if (currentChannel) {
+                        history.push(`/boards/workspace/${currentChannel}`)
+                    } else {
+                        history.goBack()
+                    }
+                }, [])
+                return <></>
+            })
             this.registry.registerProduct('/boards', GlobalHeaderIcon, 'Boards', '/plug/focalboard/go-to-current-workspace', MainApp, HeaderComponent)
+        } else {
+            this.channelHeaderButtonId = registry.registerChannelHeaderButtonAction(<FocalboardIcon/>, () => {
+                const currentChannel = store.getState().entities.channels.currentChannelId
+                window.open(`${window.location.origin}/plug/focalboard/workspace/${currentChannel}`)
```

Comment:
I believe this should be `/plugins/focalboard/...`

Reply:
The problem was with other thing, see the commit that I included.